chemprop/cli/train.py

```diff
@@ -697,81 +697,86 @@ def main(args):
         features_generators=args.features_generators, keep_h=args.keep_h, add_h=args.add_h
     )
 
-    train_data, val_data, test_data = build_splits(args, format_kwargs, featurization_kwargs)
-
     no_cv = args.num_folds == 1
-    if no_cv:
-        splits = ([train_data], [val_data], [test_data])
-    else:
-        splits = (train_data, val_data, test_data)
 
-    for fold_idx, (train_data, val_data, test_data) in enumerate(zip(*splits)):
+    for model_idx in range(args.ensemble_size):
+        
+        seed = args.seed + model_idx
 
-        train_dset, val_dset, test_dset = build_datasets(args, train_data, val_data, test_data)
+        train_data, val_data, test_data = build_splits(args, seed, format_kwargs, featurization_kwargs)
 
         if no_cv:
-            output_dir = args.output_dir
+            splits = ([train_data], [val_data], [test_data])
         else:
-            output_dir = args.output_dir / f"fold_{fold_idx}"
-            output_dir.mkdir(exist_ok=True, parents=True)
-
-        if args.save_smiles_splits:
-            save_smiles_splits(args, output_dir, train_dset, val_dset, test_dset)
+            splits = (train_data, val_data, test_data)
 
-        if args.task_type == "regression":
-            scaler = train_dset.normalize_targets()
-            val_dset.normalize_targets(scaler)
-            logger.info(f"Train data: mean = {scaler.mean_} | std = {scaler.scale_}")
-
-        train_loader = MolGraphDataLoader(train_dset, args.batch_size, args.num_workers)
-        val_loader = MolGraphDataLoader(val_dset, args.batch_size, args.num_workers, shuffle=False)
-        if test_dset is not None:
-            test_loader = MolGraphDataLoader(
-                test_dset, args.batch_size, args.num_workers, shuffle=False
-            )
-        else:
-            test_loader = None
+        for fold_idx, (train_data, val_data, test_data) in enumerate(zip(*splits)):
 
-        model = build_model(args, train_dset)
-        logger.info(model)
+            train_dset, val_dset, test_dset = build_datasets(args, train_data, val_data, test_data)
 
-        monitor_mode = "min" if model.metrics[0].minimize else "max"
-        logger.debug(f"Evaluation metric: '{model.metrics[0].alias}', mode: '{monitor_mode}'")
+            if no_cv:
+                output_dir = args.output_dir / f"model_{model_idx}"
+            else:
+                output_dir = args.output_dir / f"model_{model_idx}" / f"fold_{fold_idx}"
```

Comment:
```suggestion
            if not no_cv:
                output_dir = output_dir / f"fold_{fold_idx}"
```
and add:
```
output_dir = args.output_dir / f"model_{model_idx}"
```
before the inner loop